lib/pushmi_pullyu/preservation_queue.rb

```diff
@@ -68,10 +70,29 @@ def wait_next_item
     end
   end
 
-  def add_entity_json(entity_json)
+  def add_entity_in_timeframe(entity)
+    entity_attempts_key = "#{PushmiPullyu.options[:ingestion_prefix]}#{entity[:uuid]}"
+
     @redis.with do |connection|
-      connection.zadd @queue_name, Time.now.to_f, entity_json
+      # separate information for priority information and queue
+      deposit_attempt = connection.incr entity_attempts_key
+
+      if deposit_attempt <= PushmiPullyu.options[:ingestion_attempts]
+        connection.zadd @queue_name, Time.now.to_f + self.class.extra_wait_time(deposit_attempt),
+                        entity.slice(:uuid, :type).to_json
+      else
+        connection.del entity_attempts_key
+        raise MaxDepositAttemptsReached
+      end
+    end
+  end
+
+  def self.extra_wait_time(deposit_attempt)
+    extra = 0
+    deposit_attempt.times do |n|
+      extra += (2**n) * PushmiPullyu.options[:first_failed_wait]
     end
+    extra
   end
```

Comment:
This doesn't appear to match the stated specifications of: "doubles the time of each attempt"

With a PushmiPullyu.options[:first_failed_wait] of 10 it appears that this is the added wait time of each attempt:
```
0
10
30
70
150
310
630
1270
2550
5110
```
If doubling added wait on each attempt starting with first_failed_wait is what we're trying to accomplish it looks like this would give that:
```
def self.extra_wait_time(deposit_attempt)
  return 0 if deposit_attempt == 0

  (2**(deposit_attempt - 1)) * PushmiPullyu.options[:first_failed_wait]
end
```
The numbers vary slightly so this is a small nit in reality

Reply:
You are right!

This code should work:

```ruby
(2**deposit_attempt) * PushmiPullyu.options[:first_failed_wait]
```

Will make the changes